Index: moonlight-embedded/libgamestream/CMakeLists.txt
===================================================================
--- moonlight-embedded.orig/libgamestream/CMakeLists.txt
+++ moonlight-embedded/libgamestream/CMakeLists.txt
@@ -16,6 +16,7 @@ aux_source_directory(../third_party/moon
 aux_source_directory(../third_party/moonlight-common-c/reedsolomon MOONLIGHT_COMMON_SRC_LIST)
 
 add_library(moonlight-common SHARED ${MOONLIGHT_COMMON_SRC_LIST})
+set_property(TARGET moonlight-common PROPERTY C_STANDARD 99)
 
 add_library(gamestream SHARED ${GAMESTREAM_SRC_LIST})
 set_property(TARGET gamestream PROPERTY C_STANDARD 99)
