Index: moonlight-embedded/libgamestream/client.c
===================================================================
--- moonlight-embedded.orig/libgamestream/client.c
+++ moonlight-embedded/libgamestream/client.c
@@ -504,15 +504,12 @@ int gs_pair(PSERVER_DATA server, char* p
   char client_secret_data[16];
   RAND_bytes(client_secret_data, 16);
 
-  const ASN1_BIT_STRING *asnSignature;
-  X509_get0_signature(&asnSignature, NULL, cert);
-
   char challenge_response[16 + 256 + 16];
   char challenge_response_hash[32];
   char challenge_response_hash_enc[32];
   char challenge_response_hex[65];
   memcpy(challenge_response, challenge_response_data + hash_length, 16);
-  memcpy(challenge_response + 16, asnSignature->data, 256);
+  memcpy(challenge_response + 16, cert->signature->data, 256);
   memcpy(challenge_response + 16 + 256, client_secret_data, 16);
   if (server->serverMajorVersion >= 7)
     SHA256(challenge_response, 16 + 256 + 16, challenge_response_hash);
Index: moonlight-embedded/libgamestream/CMakeLists.txt
===================================================================
--- moonlight-embedded.orig/libgamestream/CMakeLists.txt
+++ moonlight-embedded/libgamestream/CMakeLists.txt
@@ -3,7 +3,7 @@ set(SO_VERSION 2)
 find_package(LibUUID REQUIRED)
 find_package(Threads REQUIRED)
 find_package(CURL REQUIRED)
-find_package(OpenSSL 1.0.2 REQUIRED)
+find_package(OpenSSL 1.0.1 REQUIRED)
 find_package(EXPAT REQUIRED)
 
 pkg_check_modules(AVAHI REQUIRED avahi-client)
