diff --git a/Configure b/Configure
index 744b493..590e429 100755
--- a/Configure
+++ b/Configure
@@ -1510,6 +1510,11 @@ if ($target =~ /\-icc$/)	# Intel C compiler
 		}
 	}
 
+#
+# Remove -ldl on Switch
+#
+$lflags=~s/\-ldl//;
+
 # Unlike other OSes (like Solaris, Linux, Tru64, IRIX) BSD run-time
 # linkers (tested OpenBSD, NetBSD and FreeBSD) "demand" RPATH set on
 # .so objects. Apparently application RPATH is not global and does
diff --git a/Makefile.org b/Makefile.org
index f51f0a7..7311e3b 100644
--- a/Makefile.org
+++ b/Makefile.org
@@ -137,7 +137,7 @@ FIPSCANLIB=
 
 BASEADDR=
 
-DIRS=   crypto ssl engines apps test tools
+DIRS=   crypto ssl 		# engines apps test tools
 ENGDIRS= ccgost
 SHLIBDIRS= crypto ssl
 
@@ -274,7 +274,7 @@ reflect:
 
 sub_all: build_all
 
-build_all: build_libs build_apps build_tests build_tools
+build_all: build_libs 		# build_apps build_tests build_tools
 
 build_libs: build_libcrypto build_libssl openssl.pc
 
diff --git a/crypto/ui/ui_openssl.c b/crypto/ui/ui_openssl.c
index 8a43590..c287e91 100644
--- a/crypto/ui/ui_openssl.c
+++ b/crypto/ui/ui_openssl.c
@@ -148,7 +148,7 @@
  * If unistd.h defines _POSIX_VERSION, we conclude that we are on a POSIX
  * system and have sigaction and termios.
  */
-# if defined(_POSIX_VERSION)
+# if defined(_POSIX_VERSION) && !defined(__SWITCH__)
 
 #  define SIGACTION
 #  if !defined(TERMIOS) && !defined(TERMIO) && !defined(SGTTY)
@@ -210,7 +210,8 @@
 	&& !defined(OPENSSL_SYS_MACINTOSH_CLASSIC) \
 	&& !defined(MAC_OS_GUSI_SOURCE)	\
 	&& !defined(OPENSSL_SYS_VXWORKS) \
-	&& !defined(OPENSSL_SYS_NETWARE)
+	&& !defined(OPENSSL_SYS_NETWARE) \
+    && !defined(__SWITCH__)
 #  define TERMIOS
 #  undef  TERMIO
 #  undef  SGTTY
@@ -266,7 +267,7 @@ struct IOSB {
 typedef int sig_atomic_t;
 #endif
 
-#if defined(OPENSSL_SYS_MACINTOSH_CLASSIC) || defined(MAC_OS_GUSI_SOURCE) || defined(OPENSSL_SYS_NETWARE)
+#if defined(OPENSSL_SYS_MACINTOSH_CLASSIC) || defined(MAC_OS_GUSI_SOURCE) || defined(OPENSSL_SYS_NETWARE) || defined(__SWITCH__)
 /*
  * This one needs work. As a matter of fact the code is unoperational
  * and this is only a trick to get it compiled.
